core/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -1068,6 +1069,12 @@ public void testAcquireIndexCommit() throws IOException {
             assertThat(reader.numDocs(), equalTo(flushFirst ? numDocs : 0));
         }
         commit.close();
+        // Make the global checkpoint in sync with the local checkpoint.
+        if (isPrimary) {
+            shard.getEngine().seqNoService().markAllocationIdAsInSync(shard.shardRouting.allocationId().getId(), numDocs + moreDocs - 1);
```

Comment:
why not use org.elasticsearch.index.shard.IndexShard#updateLocalCheckpointForShard ?